js_modules/dagster-ui/packages/ui-core/src/overview/OverviewSensors.tsx

```diff
@@ -94,7 +102,13 @@ export const OverviewSensors = () => {
       notifyOnNetworkStatusChange: true,
     },
   );
-  const {data, loading} = queryResultOverview;
+  const {data: currentData, loading} = queryResultOverview;
+  const data =
+    currentData ??
+    (cachedData?.workspaceOrError.__typename === 'Workspace'
+      ? (cachedData as Extract<typeof cachedData, {workspaceOrError: {__typename: 'Workspace'}}>)
+      : null);
+  useBlockTraceOnQueryResult(queryResultOverview, 'OverviewSensorsQuery');
```

Comment:
why can't we just drop the need for this query entirely?

Reply:
That will be part of a much bigger refactor were we kill all of the sub queries. FWIW this query does occur faster than the root workspace query, so technically we'd be regressing perf (at least on an uncached load) initially

Reply:
got it - can we leave some blob of comments articulating the current state and expected future state